frontend/scripts/metrica.js

```diff
@@ -0,0 +1,231 @@
+(() => {
+    let state = {
+        host: 'http://stage.metrica.fun/api/v1',
+        routes: {
+            create_visitor: '/visitors',
+            create_visit: '/visits'
+        }
+    };
+
+    const getTrackingId = () => {
+        let scripts = document.getElementsByTagName('script');
+        let index = scripts.length - 1;
+        let myScript = scripts[index];
+
+        return (myScript.src.split('tracking_id' + '=')[1] || '').split('&')[0]
+    };
+
+    const createVisitor = (tracking_number) => {
+        return fetch(state.host + state.routes.create_visitor, {
+            method: 'POST',
+            crossDomain: true,
+            headers: {
+                'Content-Type': 'application/json',
+                'x-website': tracking_number
+            },
+        })
+            .then((response) => {
+                return response.json();
+            })
+            .then((result) => {
+                console.log(result.data.token);
+                setToken(result.data.token);
+            });
+    };
+
+    const createVisit = () => {
+        return fetch(state.host + state.routes.create_visit, {
+            method: 'POST',
+            crossDomain: true,
+            headers: {
+                'Content-Type': 'application/json',
+                'x-visitor': 'Bearer ' + getToken()
+            },
+            body: JSON.stringify(getVisit())
+        });
+    };
+
+    const getUserDevice = () => {
+        let device,
+            find,
+            userAgent;
+        device = {};
+
+        window.device = device;
+
+        userAgent = window.navigator.userAgent.toLowerCase();
```

Comment:
be more accurate ((windiw.navigator || {}).userAgent || '').toLowerCase(); Or better create helper, like getProperty(window, 'navigator.userAgent', ''), that will parse by dot and retrieve safely property. And ideally it would be better to check type, as number for instance doesn't have toLowerCase method, and js will throw exception